### DISCUSSION TITLE
Trying to add bonus shares.

### DISCUSSION THREAD

**Discussion by TheRandomizer7** [ananthakumaran/paisa#166] [General]
I am trying to add bonus shares of a stock to my account.
As a perfectly reasonable approach, I did something like follows

```
Bonus Shares
Assets:Equity:<share_name>    20 <share_name> @ 0 INR
```

However,  my investment amount for the stock increased. For example say the investment amount was 2000 for 200 shares. When bonus shares were added, the investment amount changed to 2200. Which is wrong! 
Average price of stock remained unchanged. Whereas, what I wanted for it to happen was to change average price of stock and have investment amount as same. 

Any other work around is appreciated.

Edit: 
If I change 0 to 1 INR, it works as expected, it changes investment amount to 2020 and changes average price of stock. Which makes sense.

**Comment by TheRandomizer7** [comment]
Got fixed by simply closing and reopening app. But please look into this issue.

**Reply by ananthakumaran** [reply]
Could you provide a minimal journal file that reproduces the problem? I understand the problem, but a concrete journal example would make it more clear. Support for Dividend/Bonus is not perfect today (specially the investment part might not get calculated correctly)
